src/backend/commands/tablecmds.c

```diff
@@ -17136,7 +17136,7 @@ ATPExecPartRename(Relation rel,
 
 		relation_close(targetrelation, AccessExclusiveLock);
 
-		if (0 == prule->topRule->parparentoid)
+		if (prule->topRule && 0 == prule->topRule->parparentoid)
```

Comment:
I think this doesn't prevent the dereferencing [above](https://github.com/arenadata/gpdb/pull/736/files#diff-d373fb790605aded212c2e3a0fbe8c015d110077ff46c07d0a6d6647a2d7e928R17129).